collated/main/A0135768R.md

```diff
@@ -0,0 +1,1803 @@
+                sb.append(" ago") ;
+            }
+            
+            return sb.toString();
+        }
+    }
+}
+```
+###### \java\seedu\address\commons\util\FxViewUtil.java
+``` java
+ * 
+ * Contains utility methods for JavaFX views
+ */
+public class FxViewUtil {
+    
+    public static final ImmutableMap<String, NodeStyle> NODE_STYLES ;
+    
+    static {
+        
+        Builder <String, NodeStyle> builder = ImmutableMap.builder() ;
+        
+        for (NodeStyle nodeStyle : NodeStyle.values()) {
+            builder.put(nodeStyle.className, nodeStyle) ;
+        }
+        
+        NODE_STYLES = builder.build() ;
+    }
+        
+    
+    public static void removeAndAddCssClass (Node node, String classToRemove, String classToAdd) {
+        node.getStyleClass().remove(classToRemove) ;
+        node.getStyleClass().add(classToAdd) ;
+    }
+    
+    /**
+     * Sets the style of the given node with the given style CSS class if it satisfies the following rule : <p>
+     * 
+     * - The given style has a integer priority lower or equals to all other CSS classes of the same family
+     * currently applied to this node
+     * 
+     * @param node      The node to apply the style to
+     * @param style     A NodeStyle representing the CSS class that is to be applied onto the given node 
+     */
+    public static void setNodeStyle (Node node, NodeStyle style) {
+        
+        if (style == null) {
+            return ;
+        }
+        
+        String family = style.family ;
+        
+        List<String> classToRemove = new ArrayList<>() ;
+        String classToAdd = style.className ;
+        
+        for (String CssClass : node.getStyleClass()) {
+            NodeStyle nodeStyle = NODE_STYLES.get(CssClass) ;
+            
+            if (nodeStyle != null && nodeStyle.family.equals(family) && nodeStyle.priority >= style.priority) {
+                classToRemove.add(nodeStyle.className) ;
+            }
+            
+            if (nodeStyle != null && nodeStyle.family.equals(family) && nodeStyle.priority < style.priority) {
+                classToAdd = null ;
+            }
+        }
+
+        node.getStyleClass().removeAll(classToRemove) ;
+        
+        if (classToAdd != null) {
+            node.getStyleClass().add(classToAdd) ;
+        }
+        
+    }
+    
+```
+###### \java\seedu\address\logic\commands\AddCommand.java
+``` java
+     * 
+     * Convenience constructor using raw values.
+     *
+     * @throws IllegalValueException if any of the raw values are invalid
+     */
+    public AddCommand(String name, String description,String startDate,String endDate, Set<String> tags, String recurring, String repeat) throws IllegalValueException {
+
+        setRecurrenceAttributes(recurring, repeat);
+
+        if (startDate == null && endDate == null) {
+
+            setNewTaskWithDetails(name, description, tags);
+
+        } else if (startDate == null && endDate != null) {
```

Comment:
Having serious line gap issue.  This make your code too long and hard to read. Not clear at all